src/plugins/ui_core_views/cell_evaluation/evaluator.ts

```diff
@@ -188,23 +189,24 @@ export class Evaluator {
   }
 
   /**
-   * Return the position of formulas blocked by the given position
+   * Return the position of formulas blocked by the given positions
    * as well as all their dependencies.
    */
-  private getArrayFormulasBlockedBy(position: CellPosition): Iterable<CellPosition> {
-    if (!this.spreadingRelations.hasArrayFormulaResult(position)) {
-      return [];
-    }
-    const arrayFormulas = this.spreadingRelations.getFormulaPositionsSpreadingOn(position);
-    const positions = this.createEmptyPositionSet();
-    positions.addMany(arrayFormulas);
-    const arrayFormulaPosition = this.getArrayFormulaSpreadingOn(position);
-    if (arrayFormulaPosition) {
-      // ignore the formula spreading on the position. Keep only the blocked ones
-      positions.delete(arrayFormulaPosition);
+  private getArrayFormulasBlockedBy(positions: Iterable<CellPosition>): Iterable<CellPosition> {
+    const arrayFormulaPositions = this.createEmptyPositionSet();
+    for (const position of positions) {
+      if (this.spreadingRelations.hasArrayFormulaResult(position)) {
+        const arrayFormulas = this.spreadingRelations.getFormulaPositionsSpreadingOn(position);
+        arrayFormulaPositions.addMany(arrayFormulas);
+        const arrayFormulaPosition = this.getArrayFormulaSpreadingOn(position);
+        if (arrayFormulaPosition) {
+          // ignore the formula spreading on the position. Keep only the blocked ones
+          arrayFormulaPositions.delete(arrayFormulaPosition);
```

Comment:
Maybe we can remove it directly from `arrayFormulas` instead of adding it in `arrayFormulaPositions` and then remove it from the same set ?

Reply:
Problem is that `arrayFormulas` is typed as `Iterable<CellPosition>` (which I think is a good think because it makes it decoupled from the internal implementation). So we can't just remove a position from it.